src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,130 @@
+                    + id + " in database", e);
+        }
+        return Optional.ofNullable(manufacturer);
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        List<Manufacturer> allManufacturers = new ArrayList<>();
+        String readAllRequest = "SELECT * FROM manufacturers WHERE is_deleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllStatement =
+                        connection.prepareStatement(readAllRequest)) {
+            ResultSet resultSet = getAllStatement.executeQuery(readAllRequest);
+            while (resultSet.next()) {
+                allManufacturers.add(resultSetParsing(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all manufacturers from DB", e);
+        }
+        return allManufacturers;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        String updateRequest = "UPDATE manufacturers SET name = ?, country = ?"
+                + " WHERE id = ? AND is_deleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateManufacturerStatement =
+                           connection.prepareStatement(updateRequest)) {
+            updateManufacturerStatement
+                    .setString(FIRST_PARAMETER_INDEX, manufacturer.getName());
+            updateManufacturerStatement
+                    .setString(SECOND_PARAMETER_INDEX, manufacturer.getCountry());
+            updateManufacturerStatement
+                    .setLong(THIRD_PARAMETER_INDEX, manufacturer.getId());
+            int updateRows = updateManufacturerStatement.executeUpdate();
+            if (updateRows == 0) {
+                throw new DataProcessingException("There is no manufacturer with ID = "
+                        + manufacturer.getId() + " in database to update.");
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update manufacturer = "
+                    + manufacturer + " in database", e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteRequest = "UPDATE manufacturers SET is_deleted = true WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteManufacturerStatement =
+                           connection.prepareStatement(deleteRequest)) {
+            deleteManufacturerStatement.setLong(1, id);
+            return deleteManufacturerStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete manufacturers with ID = "
+                    + id + " from DB", e);
+        }
+    }
+
+    private Manufacturer resultSetParsing(ResultSet resultSet) throws SQLException {

```

Comment:
```suggestion
    private Manufacturer resultSetParsing(ResultSet resultSet) {
```
You should work the exception here, not throw it further away